File: scalar/prime.R
# TODO: Add comment
# 
# Author: Haichuan Wang(hwang154@illinois.edu)
###############################################################################

setup <- function(args='100000') {
    n<-as.integer(args[1])
    if(is.na(n)){ n <- 100000 }
    return(n)
}

#Simple trivial test of n
#no any optimizaiton, just test the worst loop case

run <- function(n=100000) {
	if(n<2) { n <- 2;}
	
	num_primes <- 0;

	for(i in 2:n) {
		limit <- sqrt(i);
		prime <- TRUE; #for 2 is prime
		j<-2; # i %% j
		while(prime && j <= limit){
			if((i %% j) ==0) { prime <- FALSE;}
			j <- j+1;
		}
		if(prime) {
			#print("prime:");print(i);
			num_primes<-num_primes+1;
		}
	}
	print(num_primes);
}


if (!exists('harness_argc')) {
    n <- setup(commandArgs(TRUE))
    run(n)
}

File: Type_I/scalar/prime/prime.c
#include <stdio.h>
#include <math.h>

int main(int argc, char* argv[]) {
    int n = 100000;
    if(argc >= 2) {
	n = atoi(argv[1]);
	if(n<2) { n = 2; }
    }

    int num_primes = 0;
    int i; double limit;
    int prime,j;
    for(i = 2; i <= n; i++) {
	limit = sqrt((double)i);
	prime = 1; //for 2 is prime
	j = 2;
	while(prime && j <= limit){
	    if((i % j) == 0) { prime = 0;}
	    j++;
	}
	if(prime) {
	    num_primes++;
	}
    }
    printf("Prime(%d)=%d\n", n, num_primes);
    return 0;
}

